include/libopencm3/gd32/f1x0/rcc.h

```diff
@@ -0,0 +1,507 @@
+#define RCC_CFGR2_PREDIV_DIV12			0xb
+#define RCC_CFGR2_PREDIV_DIV13			0xc
+#define RCC_CFGR2_PREDIV_DIV14			0xd
+#define RCC_CFGR2_PREDIV_DIV15			0xe
+#define RCC_CFGR2_PREDIV_DIV16			0xf
+
+/* --- RCC_CFGR3 values ---------------------------------------------------- */
+
+#define RCC_CFGR3_USART2SW_SHIFT		16
+#define RCC_CFGR3_USART2SW			(3 << RCC_CFGR3_USART2SW_SHIFT)
+#define RCC_CFGR3_USART2SW_PCLK			(0 << RCC_CFGR3_USART2SW_SHIFT)
+#define RCC_CFGR3_USART2SW_SYSCLK		(1 << RCC_CFGR3_USART2SW_SHIFT)
+#define RCC_CFGR3_USART2SW_LSE			(2 << RCC_CFGR3_USART2SW_SHIFT)
+#define RCC_CFGR3_USART2SW_HSI			(3 << RCC_CFGR3_USART2SW_SHIFT)
+
+#define RCC_CFGR3_ADCSW				(1 << 8)
+#define RCC_CFGR3_CECSW				(1 << 6)
+
+/* --- Variable definitions ------------------------------------------------ */
+extern uint32_t rcc_ahb_frequency;
+extern uint32_t rcc_apb1_frequency;
+extern uint32_t rcc_apb2_frequency;
+
+/* --- Function prototypes ------------------------------------------------- */
+
+enum rcc_osc {
+	RCC_PLL, RCC_HSE, RCC_HSI, RCC_LSE, RCC_LSI
+};
+
+#define _REG_BIT(base, bit)		(((base) << 5) + (bit))
+
+/* V = value line F100
+ * N = standard line F101, F102, F103
+ * C = communication line F105, F107
+ */
+enum rcc_periph_clken {
+
+	/* AHB peripherals */
+	RCC_DMA		= _REG_BIT(0x14, 0),
+	RCC_SRAM	= _REG_BIT(0x14, 2),
+	RCC_FLTF	= _REG_BIT(0x14, 4),
+	RCC_CRC		= _REG_BIT(0x14, 6),
+	RCC_GPIOA	= _REG_BIT(0x14, 17),
+	RCC_GPIOB	= _REG_BIT(0x14, 18),
+	RCC_GPIOC	= _REG_BIT(0x14, 19),
+	RCC_GPIOD	= _REG_BIT(0x14, 20),
+	RCC_GPIOF	= _REG_BIT(0x14, 22),
+	RCC_TSC		= _REG_BIT(0x14, 24),
+
+	/* APB2 peripherals */
+	RCC_SYSCFG_COMP	= _REG_BIT(0x18, 0),
+	RCC_ADC		= _REG_BIT(0x18, 9),
+	RCC_ADC1	= _REG_BIT(0x18, 9), /* Compatibility alias */
+	RCC_TIM1	= _REG_BIT(0x18, 11),
+	RCC_SPI1	= _REG_BIT(0x18, 12),
+	RCC_USART1	= _REG_BIT(0x18, 14),
+	RCC_TIM15	= _REG_BIT(0x18, 16),
+	RCC_TIM16	= _REG_BIT(0x18, 17),
+	RCC_TIM17	= _REG_BIT(0x18, 18),
+
+	/* APB1 peripherals */
+	RCC_TIM2	= _REG_BIT(0x1C, 0),
+	RCC_TIM3	= _REG_BIT(0x1C, 1),
+	RCC_TIM6	= _REG_BIT(0x1C, 4),
+	RCC_TIM14	= _REG_BIT(0x1C, 8),
+	RCC_WWDG	= _REG_BIT(0x1C, 11),
+	RCC_SPI2	= _REG_BIT(0x1C, 14),
+	RCC_SPI3	= _REG_BIT(0x1C, 15),
+	RCC_USART2	= _REG_BIT(0x1C, 17),
+	RCC_I2C1	= _REG_BIT(0x1C, 21),
+	RCC_I2C2	= _REG_BIT(0x1C, 22),
+	RCC_USB		= _REG_BIT(0x1C, 23),
+	RCC_PWR		= _REG_BIT(0x1C, 28),
+	RCC_DAC		= _REG_BIT(0x1C, 29),
+	RCC_CEC		= _REG_BIT(0x1C, 30),
+};
+
+enum rcc_periph_rst {
+
+	/* Advanced peripherals */
+	RST_BACKUPDOMAIN = _REG_BIT(0x20, 16),/* BDCR[16] */
+
+	/* AHB peripherals */
+	RST_GPIOA	= _REG_BIT(0x28, 17),
+	RST_GPIOB	= _REG_BIT(0x28, 18),
+	RST_GPIOC	= _REG_BIT(0x28, 19),
+	RST_GPIOD	= _REG_BIT(0x28, 20),
+	RST_GPIOE	= _REG_BIT(0x28, 21),
+	RST_GPIOF	= _REG_BIT(0x28, 22),
+	RST_TSC		= _REG_BIT(0x28, 24),
+
+	/* APB2 peripherals */
+	RST_SYSCFG	= _REG_BIT(0x0C, 0),
+	RST_ADC		= _REG_BIT(0x0C, 9),
+	RST_ADC1	= _REG_BIT(0x0C, 9), /* Compatibility alias */
```

Comment:
I'd actually drop the xxx_ADC defines, there's no reason to be backward compatible here.